app/routes/project/tasks/index.js

```diff
@@ -3,14 +3,22 @@ import Ember from 'ember';
 const {
   get,
   inject: { service },
-  Route
+  Route,
+  RSVP
  } = Ember;
 
 export default Route.extend({
   projectTaskBoard: service(),
 
-  model() {
-    return this.modelFor('project');
+  async model() {
+    let project = this.modelFor('project');
+    let memberships = get(project, 'organization.organizationMemberships');
+    let members = await RSVP.resolve(memberships.mapBy('member.content'));
+    return RSVP.hash({ project, members });
+  },
```

Comment:
In my opinion, this is worse than dealing with it in the previous manner. I'm actually not sure it works correctly.

`RSVP.resolve` is a helper that is intended to be used to simulate a promise (which resolves with the argument provided in the call immediately.

What might work here would be something like

```Javascript
let project = this.modelFor('project');
let memberPromises = get(project. 'organization.organizationMemberships').then((memberships) => {
  return memberships.map((membership) => get(membership, 'member'));
});

return RSVP.hash({ project, members: RSVP.all(memberPromises) });
```

This would return a fully resolved model. I don't think the above code would.

Possibly, this would work also, but I'm not sure:

```javascript
let project = this.modelFor('project');
let memberships = RSVP.all(get(project, 'organization.organizationMemberships').mapBy('member'));
```

But I'm pretty sure in that case, `.mapBy('member')` would just return null for each item.

Generally speaking, though, hashes in models are considered a smell to, and in the case of the previous approach, the whole purpose of `DS.PromiseObject`, `DS.PromiseArray` and `DS.PromiseManyArray`, which is what requests that `get` async relationships do, is to deal with the smell of promises in a computed.

These objects proxy all `get` and element requests to the `content` property, which is null until the associated promise resolves. The whole idea of having these classes in `ember-data` is so that promises "just work".

It's my opinion that this should be ok. Resolving all promises in the model hook, however, is probably the next best thing. The above code has a bug, though.

Just to reiterate:

```javascript
RSVP.resolve(x)
```

returns a promise which instantly resolves with x. Doesn't matter what that x is.


```javascript
RSVP.all(arrayOfPromises)
```

returns a promise which resolves when all promises in the array resolve.